### PR TITLE
Theme

### PR DESCRIPTION
**Pull request by DarrachBarneveld**
# ✨ Codu Pull Request 💻

Codu Logo

👉 _**Please remove the below and replace with your own values, leaving the headers where they are.**_ 👈

## Pull Request details:
Installation of next themes, tailwind animations plugin. Theme provider creation and basic theme toggler. Only nav styling bg updated. Future styles to be applied for all elements with pseudo class.

## Any Breaking changes:
- Potential client components for next 13 application. Stable in 12.
- 
## Associated Screenshots:
    
giphy

### CODE REVIEW

File: components/Theme/ThemeProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+"use client"; //Unsure if required in current next version
```
Comment (NiallJoeMaher) [inline components/Theme/ThemeProvider.tsx]: This isn't needed in this version! But that's on my plate to upgrade.

File: components/Theme/ThemeToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            as="button"
+            className="relative flex cursor-default text-base md:text-sm select-none items-center rounded-sm font-medium outline-none hover:hover:bg-neutral-200   focus:hover:bg-neutral-200  data-[disabled]:pointer-events-none data-[disabled]:opacity-50 dark:focus:bg-slate-700 text-black w-full px-4 py-2 
+"
+            onClick={() => setTheme("light")}
+          >
+            {({ active }) => (
+              <>
+                <SunIcon
+                  className="mr-2 h-4 w-4"
+                  // className={clsx(
```
Comment (NiallJoeMaher) [inline components/Theme/ThemeToggle.tsx]: Can you throw out this comment?

File: components/Theme/ThemeToggle.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import { Fragment, FunctionComponent } from "react";
+import { Menu, Transition } from "@headlessui/react";
+// import clsx from "clsx";
```
Comment (NiallJoeMaher) [inline components/Theme/ThemeToggle.tsx]: Can you remove this unused comment?

File: components/Theme/ThemeToggle.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            className="relative flex cursor-default text-base md:text-sm select-none items-center rounded-sm px-4 py-2 font-medium outline-none w-full hover:hover:bg-neutral-200   focus:hover:bg-neutral-200  data-[disabled]:pointer-events-none data-[disabled]:opacity-50 dark:focus:bg-slate-700
+            text-black
+"
+            onClick={() => setTheme("dark")}
+          >
+            {({ active }) => (
+              <>
+                <DesktopComputerIcon
+                  className="mr-2 h-4 w-4"
+                  // className={clsx(
```
Comment (NiallJoeMaher) [inline components/Theme/ThemeToggle.tsx]: Can you throw out this unused comment?

File: components/Theme/ThemeToggle.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            {({ active }) => (
+              <>
+                <MoonIcon className="mr-2 h-4 w-4" />
+                <span>Dark</span>
+              </>
+            )}
+          </Menu.Item>
+          <Menu.Item
+            as="button"
+            className="relative flex cursor-default text-base md:text-sm select-none items-center rounded-sm px-4 py-2 font-medium outline-none w-full hover:hover:bg-neutral-200   focus:hover:bg-neutral-200  data-[disabled]:pointer-events-none data-[disabled]:opacity-50 dark:focus:bg-slate-700
```
Comment (NiallJoeMaher) [inline components/Theme/ThemeToggle.tsx]: Is this a typo? 🤔 `hover:hover:bg-neutral-200`?

**Review by NiallJoeMaher — COMMENTED** [review]
Dang quick work! Thank you. Some little things to address. I'll test the functionality on dev.

**Comment by NiallJoeMaher** [comment]
Something is causing the build to break too 🤔 You might need to run `npm run build` locally to see if you can find the error.

File: components/Theme/ThemeProvider.tsx (hunk 1 above)
Comment (DarrachBarneveld) [reply components/Theme/ThemeProvider.tsx]: Noted! Will address

File: components/Theme/ThemeToggle.tsx (hunk 4 above)
Comment (DarrachBarneveld) [reply components/Theme/ThemeToggle.tsx]: Will remove all unused comments, sure thing

File: components/Theme/ThemeToggle.tsx (hunk 5 above)
Comment (DarrachBarneveld) [reply components/Theme/ThemeToggle.tsx]: Absolutely. Have you heard of https://www.npmjs.com/package/prettier-plugin-tailwindcss?
I use it but don't want my formatter to destroy anything

**Comment by DarrachBarneveld** [comment]
I think I fixed the error, but I am unsure as it requires authorization to deploy. Basically an empty prop interface can break things

File: components/Theme/ThemeToggle.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            onClick={() => setTheme("dark")}
+          >
+            <MoonIcon className="mr-2 h-4 w-4" />
+            <span>Dark</span>
+          </Menu.Item>
+          <Menu.Item
+            as="button"
+            className="relative flex cursor-default text-base md:text-sm select-none items-center rounded-sm px-4 py-2 font-medium outline-none w-full hover:bg-neutral-200   focus:hover:bg-neutral-200  data-[disabled]:pointer-events-none data-[disabled]:opacity-50 dark:focus:bg-slate-700
+            text-black"
+            onClick={() => setTheme("dark")}
```
Comment (NiallJoeMaher) [inline components/Theme/ThemeToggle.tsx]: One last question (I hope). Should this be `"system"`?

**Review by NiallJoeMaher — COMMENTED** [review]
Great work! I've never used next-themes so let me know if I'm wrong or whatever. Is it possible to set the default to dark (for now)? Until we have the other pages all done?

Cheers!

File: components/Theme/ThemeToggle.tsx (hunk 6 above)
Comment (DarrachBarneveld) [reply components/Theme/ThemeToggle.tsx]: Yeah should probably remove that for now actually. And yes in the theme provider you can set defaultTheme to dark :)

**Review by NiallJoeMaher — APPROVED** [review]
LGTM! Nice one. 🌮

**Pull request merged**